src/jit/MemoryInl.h

```diff
@@ -358,6 +439,53 @@ static void emitLoad(sljit_compiler* compiler, Instruction* instr)
         size = 8;
         break;
 #endif /* HAS_SIMD */
+#if defined(ENABLE_EXTENDED_FEATURES)
+    case ByteCode::I32AtomicLoadOpcode: {
```

Comment:
The original code reused the regular opcodes. The only thing needs to be added a `options |= MemAddress::CheckNaturalAlignment;` and a fallthrough. Is it better this way?

Reply:
I feel this way is less visually busy, but I can change it back if you wish.

Reply:
I have changed it back; I feel a macro would have to be very complex to handle the different cases of the CheckNaturalAlignment flag not always being there, and the callbacks for the various sizes and load/store variants.

When I originally changed it there were more things there, that are no longer there, so on a second look its not *too* busy.